pkgs/top-level/all-packages.nix

```diff
@@ -18421,9 +18421,28 @@ with pkgs;
 
   linbox = callPackage ../development/libraries/linbox { };
 
+  ffmpeg_4-headless = callPackage ../development/libraries/ffmpeg/4.nix {
+    inherit (darwin.apple_sdk.frameworks) Cocoa CoreMedia VideoToolbox;
+
+    sdlSupport = false;
+    vdpauSupport = false;
+    pulseaudioSupport = false;
+    libva = libva-minimal;
+  };
+
   ffmpeg_4 = callPackage ../development/libraries/ffmpeg/4.nix {
     inherit (darwin.apple_sdk.frameworks) Cocoa CoreMedia VideoToolbox;
   };
+
+  ffmpeg_5-headless = callPackage ../development/libraries/ffmpeg/5.nix {
+    inherit (darwin.apple_sdk.frameworks) Cocoa CoreMedia VideoToolbox;
+
+    sdlSupport = false;
+    vdpauSupport = false;
+    pulseaudioSupport = false;
+    libva = libva-minimal;
```

Comment:
Have you tested whether libva still works?

Reply:
Yep:
```
[AVHWDeviceContext @ 0x18484c0] Trying to use DRM render node for device 0.
[AVHWDeviceContext @ 0x18484c0] libva: VA-API version 1.15.0
[AVHWDeviceContext @ 0x18484c0] libva: Trying to open /run/opengl-driver/lib/dri/iHD_drv_video.so
[AVHWDeviceContext @ 0x18484c0] libva: Found init function __vaDriverInit_1_15
[AVHWDeviceContext @ 0x18484c0] libva: va_openDriver() returns 0
[AVHWDeviceContext @ 0x18484c0] Initialised VAAPI connection: version 1.15
[AVHWDeviceContext @ 0x18484c0] Format 0x41524742 -> bgra.
[AVHWDeviceContext @ 0x18484c0] Format 0x42475241 -> argb.
[AVHWDeviceContext @ 0x18484c0] Format 0x41424752 -> rgba.
...
...
[h264 @ 0x190d640] nal_unit_type: 8(PPS), nal_ref_idc: 3
[h264 @ 0x190d640] nal_unit_type: 5(IDR), nal_ref_idc: 3
[h264 @ 0x190d640] Format vaapi chosen by get_format().
[h264 @ 0x190d640] Format vaapi requires hwaccel initialisation.
[h264 @ 0x190d640] Considering format 0x3231564e -> nv12.
[h264 @ 0x190d640] Picked nv12 (0x3231564e) as best match for yuv420p.
[AVHWFramesContext @ 0x7f26b8049900] Created surface 0.
[AVHWFramesContext @ 0x7f26b8049900] Direct mapping possible.
[AVHWFramesContext @ 0x7f26b8049900] Created surface 0x1.
[AVHWFramesContext @ 0x7f26b8049900] Created surface 0x2.
[AVHWFramesContext @ 0x7f26b8049900] Created surface 0x3.
```

`./result-bin/bin/ffmpeg -loglevel trace -hwaccel vaapi -i ~/Videos/h264test.mp4 -f null -`

Reply:
Awesome, thank you!